src/main/scala/io/iohk/ethereum/consensus/validators/BlockWithCheckpointHeaderValidator.scala

```diff
@@ -0,0 +1,166 @@
+package io.iohk.ethereum.consensus.validators
+
+import akka.util.ByteString
+import io.iohk.ethereum.consensus.blocks.BlockGenerator
+import io.iohk.ethereum.consensus.validators.BlockHeaderError._
+import io.iohk.ethereum.{crypto, rlp}
+import io.iohk.ethereum.domain.{Address, BlockHeader}
+import io.iohk.ethereum.ledger.BloomFilter
+import io.iohk.ethereum.rlp.RLPList
+import io.iohk.ethereum.utils.{BlockchainConfig, ByteStringUtils}
+
+/** Validator specialized for the block with checkpoint
+  *
+  * @param blockchainConfig
+  */
+class BlockWithCheckpointHeaderValidator(blockchainConfig: BlockchainConfig) {
+
+  import BlockWithCheckpointHeaderValidator._
+
+  def validate(blockHeader: BlockHeader, parentHeader: BlockHeader): Either[BlockHeaderError, BlockHeaderValid] = {
+    for {
+      _ <- validateECIP1097Number(blockHeader)
+      _ <- validateCheckpointSignatures(blockHeader, parentHeader)
+      _ <- validateEmptyFields(blockHeader)
+      _ <- validateFieldsCopiedFromParent(blockHeader, parentHeader)
+      _ <- validateGasUsed(blockHeader)
+      _ <- validateTimestamp(blockHeader, parentHeader)
+    } yield BlockHeaderValid
+  }
+
+  /**
+    * Validates [[io.iohk.ethereum.domain.BlockHeader.checkpoint]] signatures
+    *
+    * @param blockHeader BlockHeader to validate.
+    * @param parentHeader BlockHeader of the parent of the block to validate.
+    * @return BlockHeader if valid, an [[io.iohk.ethereum.consensus.validators.BlockHeaderError.HeaderInvalidCheckpointSignatures]] otherwise
+    */
+  private def validateCheckpointSignatures(
```

Comment:
It won't prevent them modifying the set of sigs. I'd rather have this task done as part of the proper solution (requires changes in the checkpointing node too)

Reply:
would it prevent anyway? when the attacker has a valid set of signatures it could sort them too

Reply:
>It won't prevent them modifying the set of sigs

But wouldn't it restrict the attack to only OBFT members? (only in case OBFT produces blocks with the minimal amount of signatures)

>when the attacker has a valid set of signatures it could sort them too

But for that either:
1. The set of signatures of checkpoint blocks produced by the federation should be larger than the minimum
2. The attacker has to be able to produce valid signatures and so is a member of the federation

2. would at least reduce the attack to the federation members, 1. could maybe limited by the OBFT nodes implementation

Reply:
> But wouldn't it restrict the attack to only OBFT members? (only in case OBFT produces blocks with the minimal amount of signatures)

yes, only with that change in the OBFT node, also we should validate in Mantis that the set of signatures is minimal

Reply:
https://jira.iohk.io/browse/ETCM-174